app/gui2/src/util/opt.ts

```diff
@@ -23,3 +23,5 @@ export function isNone<T>(value: Opt<T>): value is null | undefined {
  * Helper type operator that provides the `T` type wrapped in `Opt<T>`.
  */
 export type UnwrapOpt<T> = T extends Opt<infer U> ? U : never
```

Comment:
i guess this technically works, i'd probably do `Extract<T, {}>` (because `{}` means any non-nullish value). i do think that this is probably easier to understand though

Reply:
nit: i think the `isNone` function above has an unnecessary generic

Reply:
Right. `Extract<T, {}>` would work here, but it took me a while to really understand why. Turns out there already is a type that does exactly the same thing - `NonNullable<T>`. I think i'll just remove the `UnwrapOpt` then.